### PR TITLE
Update django-modern-rpc to 1.0.0a2

### PR DESCRIPTION
**Pull request by alorence**
The goal is to test the current alpha with Kiwi and confirm the future 1.0.0 release won't introduce any regression

### CODE REVIEW

**Comment by atodorov** [comment]
```
from modernrpc.core import JSONRPC_PROTOCOL, XMLRPC_PROTOCOL

ImportError: cannot import name 'JSONRPC_PROTOCOL' from 'modernrpc.core'
```

**Comment by alorence** [comment]
I just fixed the most obvious error in urls.py, but I also noticed custom handlers were created in Kiwi. If the tests fail because of those custom handlers, I think I can update them to the new version but I will need a few more time (I should do that in the evening).

All these updates will help me writing better docs for migration, and maybe restore some deleted constant marking them as deprecated to limit the needs to update code when migration to 1.0

**Comment by atodorov** [comment]
pylint does detect some signature changes:

```
************* Module tcms.handlers
tcms/handlers.py:34:17: E1101: Super of 'KiwiTCMSJsonRpcHandler' has no 'execute_procedure' member (no-member)
tcms/handlers.py:64:17: E1101: Super of 'KiwiTCMSXmlRpcHandler' has no 'execute_procedure' member (no-member)
************* Module tcms.tests.test_handlers
tcms/tests/test_handlers.py:15:26: E1124: Argument 'entry_point' passed by position and keyword in constructor call (redundant-keyword-arg)
tcms/tests/test_handlers.py:79:26: E1124: Argument 'entry_point' passed by position and keyword in constructor call (redundant-keyword-arg)
```

File: tcms/bugs/tests/test_api.py [hunk 1]
```diff
@@ -32,7 +31,9 @@ def verify_api_with_permission(self):
         self.assertIn(self.tag, self.bug.tags.all())
 
     def verify_api_without_permission(self):
-        with self.assertRaisesRegex(ProtocolError, "403 Forbidden"):
+        with self.assertRaisesRegex(
+            XmlRPCFault, 'Authentication failed when calling "Bug.add_tag"'
+        ):
```
Comment (atodorov) [inline tcms/bugs/tests/test_api.py:36]: I would reconsider returning 200 on failure. It is non intuitive while a 403 or 500 is much more clear. This change has the potential to break existing clients which rely on the status code in order to perform some actions, e.g. a graceful error message. 

Our own RPC client is at https://github.com/kiwitcms/Kiwi/blob/master/tcms/static/js/jsonrpc.js#L23. I need to review in more details if the beckend change will not break the FE client but essentially we want to be able to show an alert to the user when something fails.

Comment (atodorov) [reply tcms/bugs/tests/test_api.py:36]: Can you point me to the actual commit/commits or place in the code which implement this change? 

We do have quite a lot of API clients and I'd like to override this change and revert to the old behavior, e.g. by supplying a custom XMLRpcHandler. We've already got `KiwiTCMSXmlRpcHandler` so overriding a few more methods should be relatively easy.  Then we could think about phasing out the old behavior properly on our side.

Comment (atodorov) [reply tcms/bugs/tests/test_api.py:36]: I am back to reviewing this PR. 

From what I can tell our RPC clients don't really depend on the 403 return code and after doing some testing it looks like they are working fine. I am going ahead with this change as-is.

**Review by atodorov — APPROVED** [review]

**Pull request merged**